main.py

```diff
@@ -39,65 +39,64 @@ def _run(self):
         for submission in self.reddit.subreddit(self.config.SUBREDDIT).stream.submissions():
             if self.database.known_submission(submission) is True:
                 continue
+            else:
+                self.database.save_submission(submission)
 
             if submission.is_self is True:
                 if '[discussion]' not in str(submission.title).lower():
-                    self._archive_text_post_without_discussion_tag(submission)
-                else:
-                    self.database.save_submission(submission)
+                    self._remove_text_post_without_discussion_tag(submission)
                 continue
 
             url = re.search("(?P<url>https?://[^\s]+)", submission.url).group("url")
 
             parsed_url = urlparse(url)
-            youtube_service = self.youtube.get_service_result(parsed_url)
 
-            if youtube_service.exceeds_threshold is True:
-                self._archive_submission_exceeding_threshold(submission, youtube_service.service_name,
-                                                             youtube_service.threshold, youtube_service.listeners_count)
-                continue
+            artist_name = self._get_artist_name_from_submission_title(submission.title)
 
-            try:
-                artist_name = self._get_artist_name_from_submission_title(submission.title)
-            except LookupError:
-                self._reply_with_sticky_post(submission, templates.cant_find_artist)
+            youtube_service = self.youtube.get_service_result(parsed_url)
 
+            if youtube_service.exceeds_threshold is True:
+                self._perform_exceeds_threshold_mod_action(submission, youtube_service)
                 continue
 
             try:
                 last_fm_service = self.last_fm.get_service_result(artist_name)
+                if last_fm_service.exceeds_threshold is True:
+                    self._perform_exceeds_threshold_mod_action(submission, last_fm_service)
+                    continue
             except pylast.WSError:
-                continue
```

Comment:
We previously failed out if LastFM couldn't connect.  Instead, I think we continue on and just know we may have missed a check somewhere.

Reply:
Definitely possible (and arguably likely for sure).  But in that case we simply won't get the nice LastFM lookup reply post but will have still done anything we could at the time.